Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h

```diff
@@ -28,27 +28,35 @@
 #if PLATFORM(COCOA)
 
 #include <wtf/ArgumentCoder.h>
+#include <wtf/RetainPtr.h>
+#include <wtf/Vector.h>
 
 namespace WebKit {
 
+class CoreIPCNSCFObject;
+
 class CoreIPCDictionary {
 public:
-    CoreIPCDictionary(NSDictionary *dictionary)
-        : m_dictionary(dictionary)
-    {
-    }
+    CoreIPCDictionary(NSDictionary *);
 
     CoreIPCDictionary(RetainPtr<NSDictionary>&& dictionary)
-        : m_dictionary(WTFMove(dictionary))
+        : CoreIPCDictionary(dictionary.get())
     {
     }
 
-    RetainPtr<id> toID() { return m_dictionary; }
+    RetainPtr<id> toID() const;
 
 private:
     friend struct IPC::ArgumentCoder<CoreIPCDictionary, void>;
 
-    IPC::CoreIPCRetainPtr<NSDictionary> m_dictionary;
+    using ValueType = Vector<std::pair<std::unique_ptr<CoreIPCNSCFObject>, std::unique_ptr<CoreIPCNSCFObject>>>;
```

Comment:
Ditto.  Could use WTF::KeyValuePair, don't see need for unique_ptr.  Use UniqueRef if you have to have another allocation, because there's certainly no need for a null state.

It may also be possible that we only need to support keys that are Strings.

Reply:
Today, we absolutely do NSDictionaries with arbitrary keys.
I don't think this patch is the right place to rethink that (going for zero behavior change)